### PR TITLE
[tests-only][full-ci] add test to check activities of destination file and folder by moving a file using file-id

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds tests to check activities of destination file and folder after moving a file using it's file-id into a folder.
``` feature
Scenario: check activities of destination folder after moving a file into it using file-id
Scenario: check activities of destinaton file and folder after moving a 0 byte file using file-id
Scenario: check activities of destinaton file and folder after moving a file by renaming destination file using file-id
Scenario: check activities of destinaton file and folder after moving a 0 byte file by renaming destination file using file-id
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/9543
- https://github.com/owncloud/ocis/issues/9543#issuecomment-2431703972

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -2065,9 +2065,9 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
 		$sourceFile = \explode("/", $sourceFile);
 		$sourceFile = \end($sourceFile);
 		$destinationFile = \trim($destinationFile, "/");
-		if ($actionType === 'copied' || $actionType === 'moved') {
+		if ($actionType === 'copied') {
 			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
-		} else {
+		} elseif ($actionType === 'renamed' || $actionType === 'moved') {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:2070]: was it required? and why

Comment (prashant-gurung899) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: Yes, because for following scenario we have to directly rename the file during the MOVE request:
``` feature
Scenario: check activities of destinaton file and folder after moving a file by renaming destination file using file-id
```
In this particular step:
``` feature 
And user "Alice" has moved file "textfile.txt" into "FOLDER/renamed.txt" inside space "Personal" using file-id "<<FILEID>>" 
```

If we didn't do it then it would return `204` instead of `201`
See: https://drone.owncloud.com/owncloud/ocis/40432/38/5

cc @saw-jan

Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: hmm, I think there should be another better way. I know these steps are confusing.

Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: Actually, what's happening is that: 
What we already have is:
```php
...
if ($actionType === 'copied' || $actionType === 'moved') {
			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
		} else {
			$fileDestination = $destinationFile;
		}
...
```
And for the steps like:
```
    And user "Alice" has copied file "textfile.txt" into "newFolder" inside space "Personal" using file-id "<<FILEID>>"
```
we are not appending the `file name` after the container i.e., `folder name` (like this: newFolder/textfile.txt). So, it gives:
```
$fileDestination: "newFolder/textfile.txt" and
$headers: Destination="https://localhost:9200/remote.php/dav/spaces/98b0f059-cf07-4bac-8528-bcaed059edcb$6ab70a1e-4a57-4542-9f76-eb9982a37da5/newFolder/textfile.txt"
```

But for scenarios like:
```
    And user "Alice" has moved file "textfile.txt" into "New Folder/textfile.txt" inside space "Personal" using file-id "<<FILEID>>"
```
we are already appending the `file name` after the container i.e., `folder name` (like this: New Folder/textfile.txt). So it gives:
```
$fileDestination = "New%20Folder/textfile.txt/textfile.txt" and
$headers: Destination="https://localhost:9200/remote.php/dav/spaces/98b0f059-cf07-4bac-8528-bcaed059edcb$6826391a-448e-42b1-9ef4-cfb39c0c4ac4/New%20Folder/textfile.txt/textfile.txt"
```
which is not expected.

So 
EITHER
 we can modify the `copy` steps appending the `file name` after container and we can totally omit above mentioned code snippet
OR
we can modify the `move` steps removing the `file name` after container and we can have the existing above mentioned code snippet
OR
we can go with changes made in this PR
OR we should dig more to find another solution

Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: I think, modifying the `copy` steps and omitting the snippet 
```php
if ($actionType === 'copied' || 'moved') {
			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
		} else {
			$fileDestination = $destinationFile;
		}
```
will be better choice because in other test cases also, the `file name` is appended after its container like:
https://github.com/owncloud/ocis/blob/5fc891ce94686bd89f5e6b082ca2f85c80f478f4/tests/acceptance/features/coreApiFavorites/favorites.feature#L154

Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: IMO, the scenario steps need to be written clearly.
Example: 
```feature
# append source <file> to the destination path (<folder>/<file>)
... (copies|moves) a file "<file>" into folder "<folder>" inside space "<space>"

# do not append source <file> to the destination path (<newfile>)
... (copies|moves|renames) a file "<file>" to "<newfile>" inside space "<space>"

# do not append source <file> to the destination path (<folder>/<newfile>)
... (copies|moves|renames) a file "<file>" to "<folder>/<newfile>" inside space "<space>"
```

Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: IMO, this would be better
```feature
# append source <file> to the destination path (<folder>/<file>)
... (copies|moves) a file "<file>" into folder "<folder>" inside space "<space>"

# do not append source <file> to the destination path (<newfile>)
... renames a file "<file>" to "<newfile>" inside space "<space>"

# do not append source <file> to the destination path (<folder>/<newfile>)
... renames a file "<file>" to "<folder>/<newfile>" inside space "<space>"
```

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 2]
```diff
@@ -472,3 +472,809 @@ Feature: check activities
...
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        }
+      }
+      """
+    When user "Alice" lists the activities of file "FOLDER" from space "Personal" using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:701]:
```suggestion
    When user "Alice" lists the activities of folder "FOLDER" from space "Personal" using the Graph API
```

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 3]
```diff
@@ -472,3 +472,809 @@ Feature: check activities
...
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:929]:
```suggestion
    When user "Alice" lists the activities of folder "FOLDER" from space "Personal" using the Graph API
```

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 4]
```diff
@@ -472,3 +472,809 @@ Feature: check activities
...
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:1166]:
```suggestion
    When user "Alice" lists the activities of folder "FOLDER" from space "Personal" using the Graph API
```

File: tests/acceptance/bootstrap/SpacesContext.php (hunk 1 above)
Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php]: @saw-jan Refactored the steps. Please have a look.

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 5]
```diff
@@ -2062,13 +2062,12 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
...
-		$sourceFile = \end($sourceFile);
 		$destinationFile = \trim($destinationFile, "/");
-		if ($actionType === 'copied' || $actionType === 'moved') {
-			$fileDestination = $this->escapePath($destinationFile) . '/' . $this->escapePath($sourceFile);
-		} else {
-			$fileDestination = $destinationFile;
+
+		$fileDestination = $this->escapePath($destinationFile);
+
+		if ($actionType !== 'renamed' && !strpos($destinationFile, "/")) {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:2069]: this generally looks good but IMO we better have separate step for rename action

Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php:2069]: Yeah. Sounds better.. Did it just now

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 6]
```diff
@@ -2062,13 +2062,12 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
...
+			$fileDestination .= '/' . $this->escapePath($sourceFile);
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:2070]: we should be able to revert these changes now

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 7]
```diff
@@ -2093,6 +2092,43 @@ public function userHasCopiedOrMovedFileInsideSpaceUsingFileId(
 		);
 	}
 
+	/**
+	 * @Given /^user "([^"]*)" has renamed file with file-id "([^"]*)" to "([^"]*)" inside space "([^"]*)"$/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:2096]:
```suggestion
	 * @Given /^user "([^"]*)" has renamed file with id "([^"]*)" to "([^"]*)" inside space "([^"]*)"$/
```

File: tests/acceptance/bootstrap/SpacesContext.php (hunk 6 above)
Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php:2070]: I think we will be needing:
```php
if (!strpos($destinationFile, "/")) {
			$fileDestination .= '/' . $this->escapePath($sourceFile);
		}
```
for steps like:
```gherkin
    And user "Alice" has moved file "textfile.txt" into "FOLDER/renamed.txt" inside space "Personal" using file-id "<<FILEID>>"
```
because we have to append destination file name here

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 8]
```diff
@@ -472,3 +472,809 @@ Feature: check activities
         }
       }
       """
+
+
+  Scenario: check activities of destination folder after moving a file into it using file-id
+    Given user "Alice" has uploaded file with content "ownCloud test text file" to "textfile.txt"
+    And we save it into "FILEID"
+    And user "Alice" has created folder "FOLDER"
+    And user "Alice" has moved file "textfile.txt" into "FOLDER" inside space "Personal" using file-id "<<FILEID>>"
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:481]: how about these steps?
```feature
    And user "Alice" has (copied|moved) a file with id "<<FILEID>>" as "textfile.txt" into folder "FOLDER" of space "Personal"
```

File: tests/acceptance/bootstrap/SpacesContext.php (hunk 1 above)
Comment (amrita-shrestha) [reply tests/acceptance/bootstrap/SpacesContext.php]: @pradip did you make a separate issue / add in an existing issue for modifying steps as commented above

Comment (anon-pradip) [reply tests/acceptance/bootstrap/SpacesContext.php]: > @pradip did you make a separate issue / add in an existing issue for modifying steps as commented above

I made a separate new commit for modifying the steps in this PR  and I am pushing changes that might touch previous changes in this new commit

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature (hunk 8 above)
Comment (anon-pradip) [reply tests/acceptance/features/apiActivities/activitiesByFileId.feature]: Sounds awesome. Leaning to this idea then :slightly_smiling_face:

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 9]
```diff
@@ -472,3 +472,809 @@ Feature: check activities
         }
       }
       """
+
+
+  Scenario: check activities of destination folder after moving a file into it using file-id
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:477]: This scenario can be merged with the one above
https://github.com/owncloud/ocis/blob/3847f46dedd7c5e88db6da335ee7a8b4dcfc38f4/tests/acceptance/features/apiActivities/activitiesByFileId.feature#L364

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 10]
```diff
@@ -472,3 +472,821 @@ Feature: check activities
...
+                          "required": ["resource"],
+                          "properties": {
+                            "resource": {
+                              "type": "object",
+                              "required": ["id","name"],
+                              "properties": {
+                                "id": {
+                                  "type": "string",
+                                  "pattern": "^%file_id_pattern%$"
+                                },
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:512]: you can only check the `name`
```suggestion
```

File: tests/acceptance/features/apiActivities/activitiesByFileId.feature [hunk 11]
```diff
@@ -365,7 +365,7 @@ Feature: check activities
     Given user "Alice" has uploaded file with content "ownCloud test text file" to "textfile.txt"
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activitiesByFileId.feature:365]: reword the scenario title similar to the scenarios below

**Review by saw-jan — APPROVED** [review]
